src/main/java/com/github/hokkaydo/eplbot/module/code/CodeCommand.java

```diff
@@ -0,0 +1,208 @@
+                })
+                .exceptionally(t -> {
+                    t.printStackTrace();
+                    return null;
+                });
+        
+    }
+    @Override
+    public void onModalInteraction(@NotNull ModalInteractionEvent event) {
+        if(event.getInteraction().getType() != InteractionType.MODAL_SUBMIT || !event.getModalId().contains("Code")) return;
+        Optional<ModalMapping> lang = Optional.ofNullable(event.getInteraction().getValue("language"));
+        Optional<ModalMapping> body = Optional.ofNullable(event.getInteraction().getValue("body"));
+        Guild guild = event.getGuild();
+        if(lang.isEmpty() || body.isEmpty() || guild == null) return;
+
+        Integer runTimeout = Config.getGuildVariable(guild.getIdLong(), "COMMAND_CODE_TIMELIMIT");
+        event.getInteraction().reply("Processing since: <t:" + Instant.now().getEpochSecond() + ":R>").queue();
+
+        String languageOption = Objects.requireNonNull(lang.get().getAsString());
+        String bodyStr = Objects.requireNonNull(body.get().getAsString());
+        Runner runner = RUNNER_MAP.get(languageOption);
+        performSubmit(event.getMessageChannel(),bodyStr, languageOption);
+        performResponse(event.getMessageChannel(),runner.run(bodyStr, runTimeout));
+    }
+    private void performSubmit(MessageChannel textChannel, String bodyStr,String codeName){
+        if (!validateMessageLength(bodyStr)){
+            sendWrappedCodeAsText(textChannel,bodyStr,codeName);
+            return;
+        }
+        File submitted = createWrappedCodeAsFile(bodyStr);
+        if (submitted == null){
+            textChannel.sendMessage(Strings.getString("COMMAND_CODE_COULDNT_WRITE_THE_FILE")).queue();
+            return;
+        }
+        if (!validateFileSize(submitted)){
+            sendSubmittedAsFile(textChannel,submitted);
+            return;
+        }
+        textChannel.sendMessage(Strings.getString("COMMAND_CODE_EXCEEDED_FILE_SIZE")).queue();
+    }
+    private void performResponse(MessageChannel textChannel, String result){
+        if (!validateMessageLength(result)){
+            textChannel.sendMessage("`"+result+"`").queue();
+            return;
+        }
+        File submitted = createResponseAsFile(result);
+        if (submitted == null){
+            textChannel.sendMessage(Strings.getString("COMMAND_CODE_COULDNT_WRITE_THE_FILE")).queue();
+            return;
+        }
+        if (!validateFileSize(submitted)){
+            sendResponseAsFile(textChannel,submitted);
+            return;
+        }
+        textChannel.sendMessage(Strings.getString("COMMAND_CODE_EXCEEDED_FILE_SIZE")).queue();
+    }
+    private void sendWrappedCodeAsText(MessageChannel textChannel, String bodyStr, String codeName){
+        textChannel.sendMessage("```"+codeName.toLowerCase()+"\n"+bodyStr+"\n```").queue();
+    }
+    private boolean validateMessageLength(String content){
+        return content.length() >= 2000;
```

Comment:
Function return is always inverted when using 
=> content.length < 2000 and don't use `!`